app/src/main/java/org/mozilla/focus/components/DownloadCompleteReceiver.java

```diff
@@ -57,15 +59,21 @@ public void onReceive(Context context, Intent intent) {
                     downloadInfo.setDownloadId(downloadId);
                     downloadInfo.setFileName(fileName);
                     downloadInfo.setFileUri(localUriStr);
-                    DownloadInfoManager.getInstance().update(downloadInfo,null);
+                    DownloadInfoManager.getInstance().updateByDownloadId(downloadInfo,null);
                     if (downloadedFile.exists() && downloadedFile.canWrite()) {
                         RelocateService.startActionMove(context,
                                 downloadId,
                                 downloadedFile,
                                 mediaType);
                     }
+
+                    String completedStr = context.getString(R.string.download_completed);
+                    Toast.makeText(context,fileName+" "+completedStr,Toast.LENGTH_SHORT).show();
                 }
             }
+        }else {
```

Comment:
Last mile: Let's also notify Adapter with a new delete item method to deal with the case when the user cancels with Download Manager Toast...well this is an edge case so I guess we can skip this for now. Once we merge this I will file another bug for this and I might triage it as P2.